src/simplewallet/simplewallet.cpp

```diff
@@ -5765,6 +5769,56 @@ bool simple_wallet::get_tx_key(const std::vector<std::string> &args_)
   }
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::set_tx_key(const std::vector<std::string> &args_)
+{
+  std::vector<std::string> local_args = args_;
+
+  if(local_args.size() != 2) {
+    fail_msg_writer() << tr("usage: set_tx_key <txid> <tx_key>");
+    return true;
+  }
+
+  crypto::hash txid;
+  if (!epee::string_tools::hex_to_pod(local_args[0], txid))
+  {
+    fail_msg_writer() << tr("failed to parse txid");
+    return true;
+  }
+
+  crypto::secret_key tx_key;
+  if (!epee::string_tools::hex_to_pod(local_args[1].substr(0, 64), tx_key))
```

Comment:
That throws if the string is less than 64: need to catch.

Reply:
I don't see `hex_to_pod` throwing; it just returns `false` when the string size is wrong:

https://github.com/monero-project/monero/blob/master/contrib/epee/include/string_tools.h#L343
```c++
  template<class t_pod_type>
  bool hex_to_pod(const std::string& hex_str, t_pod_type& s)
  {
    ...
    if(sizeof(s)*2 != hex_str.size())
      return false;
    ...
  }
```

Reply:
True, thanks.